image/tarexport/os_path.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2009 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Code in this file is a modified version of go stdlib;
+// https://cs.opensource.google/go/go/+/refs/tags/go1.23.4:src/os/path.go;l=19
```

Comment:
Would be nice if we split this into two separate commits:

1. Import the function 1:1 from the go source
2. Adjust the code with our modifications

Like https://github.com/moby/moby/commit/56bb143a4d55f73a1e2ede192d782369df14abe5 + https://github.com/moby/moby/commit/5841ed4e5e0ea0a9b78fa6759cd9b5929463c313

Reply:
Not a bad suggestion, it makes it easier indeed to see "what did we change". With the small note that of course the first copy would need _slight_ changes (we can't use the `filepathlite` as it's internal (but `filepath.VolumeName` is the same); https://cs.opensource.google/go/go/+/refs/tags/go1.23.4:src/os/path.go;l=47

And need to add the `os.` before exported functions.

I'll work on that 👍

Reply:
I don't think we necessarily need to adjust the first copy. While it makes the first commit not buildable, I think it's not a big problem. In practice it would only be a problem when bisecting, but since `git bisect skip` exists... Personally I wouldn't worry much about it.

Reply:
OK; updated 👍 

(CI failure is a current incident with Docker Hub)

Reply:
The changes to `image/tarexport/os_path.go` in 17a0fc5589d3d69f0cf8d7c3b4916847585f5ed0 should also go into a separate commit after the non-buildable one. So in future it can be cherry-picked on its own if the upstream code changes and we need to rebase our forked copy.

Reply:
Ah! Done 👍